atch {
          case e: Exception =>
            failedQueries += transformedQuery
            println(s"Failed to execute query: $transformedQuery\nError: ${e.getMessage}")
            
            if (e.getMessage.contains("SCT_TEST") && e.getMessage.contains("duplicate")) {
              println("Duplicate found in SCT_TEST, stopping execution.")
              conn_rss.rollback()
              throw new RuntimeException("Secondary index uniqueness violation in SCT_TEST, execution stopped.")
            }
